common/app/routes/Challenges/redux/index.js

```diff
@@ -311,9 +319,20 @@ export default combineReducers(
         ...state,
         output: (state.output || '') + output
       }),
-
-      [types.openBugModal]: state => ({ ...state, isBugOpen: true }),
-      [types.closeBugModal]: state => ({ ...state, isBugOpen: false })
+      [types.openBugModal]: state => ({
+        ...state,
+        isModalOpen: true,
+        modalType: 'bug'
+      }),
+      [types.openHelpModal]: state => ({
+        ...state,
+        isModalOpen: true,
+        modalType: 'help'
+      }),
```

Comment:
This is an unnecessary complication and couples both modals needlessly.

Instead, give each modal it's own state. `isBugModalOpen` and `isHelpModalOpen`. This removes the need for the `Challenges/Modals` since this is ultimately just a state switch. 

With this done, you can each individual modal open or close without having to know anything about each other.

Reply:
Should there also be two separate epics to handle actions originating from separate modals?

Reply:
Do you mean epic files? I'd make four epics, one for each effect. I believe that the effects are close enough in feature that they should exist in one file and be combined similiar to https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/common/app/routes/Challenges/redux/code-storage-epic.js

Reply:
Done